### DISCUSSION TITLE
How do I restart the app automatically after an update?

### DISCUSSION THREAD

**Discussion by murnifine** [tauri-apps/tauri#10383] [Q&A]
I implemented the updater plugin in windows, then I show a popup that updates are available, after that I click update, then the application closes and I have to open it manually again.

Is there any way or function to open automatically after the update?

**Comment by FabianLars** [comment]
It's supposed to re-open automatically. Can you share the output of the `tauri info` command and the (rust) version of the updater plugin?

**Reply by murnifine** [reply]
There are two problems I encountered
1. After updating the application does not open automatically
2. After I Uninstall, the application still exists

I am using the .msi installer 

```
[✔] Environment
    - OS: Windows 10.0.22621 X64
    ✔ WebView2: 126.0.2592.113
    ✔ MSVC:
        - Visual Studio Build Tools 2022
        - Visual Studio Community 2022
    ✔ rustc: 1.79.0 (129f3b996 2024-06-10)
    ✔ cargo: 1.79.0 (ffa9cf99a 2024-06-03)
    ✔ rustup: 1.27.1 (54dd3d00f 2024-04-24)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 18.20.2
    - pnpm: 8.15.3
    - npm: 10.8.1
    - bun: 1.1.4

[-] Packages
    - tauri [RUST]: 2.0.0-beta.24
    - tauri-build [RUST]: 2.0.0-beta.19
    - wry [RUST]: 0.41.0
    - tao [RUST]: 0.28.1
    - @tauri-apps/api [NPM]: 2.0.0-beta.15
    - @tauri-apps/cli [NPM]: 2.0.0-beta.22

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../out
    - devUrl: http://localhost:3000/
    - framework: React (Next.js)
    - bundler: Webpack
```

**Reply by murnifine** [reply]
```json
"@tauri-apps/plugin-updater": "^2.0.0-beta.7"
```

```toml
[target."cfg(not(any(target_os = \"android\", target_os = \"ios\")))".dependencies]
tauri-plugin-updater = "2.0.0-beta.11"
```

**Reply by murnifine** [reply]
it says windows 10, but actually I use windows 11

**Reply by FabianLars** [reply]
> After updating the application does not open automatically
> After I Uninstall, the application still exists

This sounds like the first instance was either not an _installed_ app instance, or what you installed does not match the updater package. So for example, if you install your app via .msi the updater endpoint must serve the .msi (or .msi.zip), or the other way around for nsis (setup.exe)

> it says windows 10, but actually I use windows 11

Funnily enough, that's how windows 11 reports itself.

**Comment by matthiasg** [comment]
How exactly is the restart supposed to work. I have been trying here on Mac and it quits, it updates the .app contents, but I have yet have it restart.
